aws/resource_aws_ssm_document.go

```diff
@@ -171,17 +166,25 @@ func resourceAwsSsmDocument() *schema.Resource {
 			"target_type": {
 				Type:     schema.TypeString,
 				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^\/[\w\.\-\:\/]*$`), ""),
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16628

```suggestion
					validation.StringMatch(regexp.MustCompile(`^\/[\w\.\-\:\/]*$`), "must contain a forward slash optionally followed by a resource type such as AWS::EC2::Instance"),
```